### PR TITLE
feat(#454): Added deployment config to Feedback SPA

### LINKED ISSUE #454: Add deployment support to all SPAs
## Is your feature request related to a problem? Please describe
All individual SPAs need to have deployment configuration enabled for easy deployments to various environments

## Describe the solution you'd like
Add a deployment script and prime all SPAs for easy deployment.
- [x] Home
- [x] Feedback
- [x] Notfication
- [x] User Profile

### PR DESCRIPTION
**Pull request by sayak-sarkar**
# Partially Resolves
#454

# Explain the feature/fix
Added deployment script and primed all environment config for deployment.

## Does this PR introduce a breaking change
No

## Notes
the deployment script is still pretty nascent. It would be further improved and probably merged into OP CLI in future PRs.

### CODE REVIEW

**Comment by hybridx** [comment]
@sayak-sarkar @deshmukhmayur @riginoommen  Should we be using `'` and .ts files and not `"`?
Because `"` throws lint error for me

File: packages/feedback-spa/src/app/feedback/feedback-item/feedback-item.component.ts [hunk 1]
```diff
@@ -1,21 +1,19 @@
-import { Component, OnInit, Input, Output, EventEmitter} from '@angular/core';
+import { Component, OnInit, Input, Output, EventEmitter } from "@angular/core";
 
 @Component({
-  selector: 'op-feedback-item',
-  templateUrl: './feedback-item.component.html',
-  styleUrls: ['./feedback-item.component.scss']
+  selector: "op-feedback-item",
```
Comment (riginoommen) [inline packages/feedback-spa/src/app/feedback/feedback-item/feedback-item.component.ts:4]:
```suggestion
  selector: 'op-feedback-item',
```

File: packages/feedback-spa/src/app/feedback/feedback-item/feedback-item.component.ts [hunk 2]
```diff
@@ -1,21 +1,19 @@
...
+  templateUrl: "./feedback-item.component.html",
+  styleUrls: ["./feedback-item.component.scss"],
```
Comment (riginoommen) [inline packages/feedback-spa/src/app/feedback/feedback-item/feedback-item.component.ts:6]:
```suggestion
  styleUrls: ['./feedback-item.component.scss'],
```

**Comment by sayak-sarkar** [comment]
@hybridx I kept trying to use `'` but I think `editorconfig` conf kept overwriting it. Need to check that in the future.

File: packages/feedback-spa/src/app/feedback/home/feedback-home.component.ts [hunk 3]
```diff
@@ -1,20 +1,19 @@
-import { Component, OnInit, ViewChild, ElementRef } from '@angular/core';
-import { FeedbackService } from '../feedback.service';
-import { environment } from '../../../environments/environment';
-import { options } from '../../../mocks/options';
+import { Component, OnInit, ViewChild, ElementRef } from "@angular/core";
+import { FeedbackService } from "../feedback.service";
+import { environment } from "../../../environments/environment";
+import { options } from "../../../mocks/options";
```
Comment (riginoommen) [inline packages/feedback-spa/src/app/feedback/home/feedback-home.component.ts:4]:
```suggestion
import { Component, OnInit, ViewChild, ElementRef } from '@angular/core';
import { FeedbackService } from '../feedback.service';
import { environment } from '../../../environments/environment';
import { options } from '../../../mocks/options';
```

File: packages/feedback-spa/src/app/feedback/home/feedback-home.component.ts [hunk 4]
```diff
@@ -1,20 +1,19 @@
...
-@Component( {
-  selector: 'op-feedback-home',
-  templateUrl: './feedback-home.component.html',
-  styleUrls: [ './feedback-home.component.scss' ]
-} )
+@Component({
+  selector: "op-feedback-home",
+  templateUrl: "./feedback-home.component.html",
+  styleUrls: ["./feedback-home.component.scss"],
+})
```
Comment (riginoommen) [inline packages/feedback-spa/src/app/feedback/home/feedback-home.component.ts:10]:
```suggestion
@Component({
  selector: 'op-feedback-home',
  templateUrl: './feedback-home.component.html',
  styleUrls: ['./feedback-home.component.scss'],
})
```

File: packages/feedback-spa/src/app/feedback/home/feedback-home.component.ts [hunk 5]
```diff
@@ -1,20 +1,19 @@
...
 export class FeedbackHomeComponent implements OnInit {
-
-  @ViewChild( 'fiModalTrigger' ) fiModalTrigger: ElementRef;
+  @ViewChild("fiModalTrigger") fiModalTrigger: ElementRef;
```
Comment (riginoommen) [inline packages/feedback-spa/src/app/feedback/home/feedback-home.component.ts:12]:
```suggestion
  @ViewChild('fiModalTrigger') fiModalTrigger: ElementRef;
```

File: packages/feedback-spa/src/app/feedback/home/feedback-home.component.ts [hunk 6]
```diff
@@ -34,140 +33,165 @@ export class FeedbackHomeComponent implements OnInit {
...
+          this.allFeedback = allFeedback.map((fb) => {
+            const fbObj = { ...fb };
+            const fbIssue = jiraIssues.filter((issue) => {
+              if (issue.ticketID === fb.ticketID) {
+                return issue;
+              }
+            })[0];
+            if (fbIssue) {
+              fbObj.state = fbIssue.status.name;
+              fbObj.assignees = fbIssue.assignee.name.split(",");
```
Comment (riginoommen) [inline packages/feedback-spa/src/app/feedback/home/feedback-home.component.ts:52]:
```suggestion
              fbObj.assignees = fbIssue.assignee.name.split(',');
```

File: packages/feedback-spa/src/app/feedback/home/feedback-home.component.ts [hunk 7]
```diff
@@ -34,140 +33,165 @@ export class FeedbackHomeComponent implements OnInit {
...
+          this.allFeedback = this.allFeedback.filter((fb) => {
+            if (!fb.state) {
+              fb.state = "closed";
+            }
+            if (!fb.module) {
+              fb.module = {
+                name: "general",
+              };
+            }
+            return fb;
```
Comment (riginoommen) [inline packages/feedback-spa/src/app/feedback/home/feedback-home.component.ts:65]:
```suggestion
          });
          this.allFeedback = this.allFeedback.filter((fb) => {
            if (!fb.state) {
              fb.state = 'closed';
            }
            if (!fb.module) {
              fb.module = {
                name: 'general',
              };
            }
            return fb;
```

File: packages/feedback-spa/src/app/feedback/home/feedback-home.component.ts [hunk 8]
```diff
@@ -34,140 +33,165 @@ export class FeedbackHomeComponent implements OnInit {
...
+          });
+          this.stats.push({
+            _id: 1,
+            labelName: "All",
+            labelSize: "24",
+            countNumber: this.totalFeedbackCount.toString(),
+            countSize: "42",
+            themeColor: "#219be0",
+            totalCountNumber: this.totalFeedbackCount.toString(),
+          });
```
Comment (riginoommen) [inline packages/feedback-spa/src/app/feedback/home/feedback-home.component.ts:102]:
```suggestion
          this.setFeedbackCount();
          this.totalFeedbackCount = this.allFeedback.length;
          this.bugsCount = this.allFeedback.filter(
            (feedback) => feedback.feedbackType === 'Bug'
          ).length;
          this.feedbackCount = this.allFeedback.filter(
            (feedback) => feedback.feedbackType === 'Feedback'
          ).length;
          this.stats.push({
            _id: 3,
            labelName: 'Bugs',
            labelSize: '24',
            countNumber: this.bugsCount.toString(),
            countSize: '42',
            themeColor: '#219be0',
            totalCountNumber: this.bugsCount.toString(),
          });
          this.stats.push({
            _id: 2,
            labelName: 'Feedback',
            labelSize: '24',
            countNumber: this.feedbackCount.toString(),
            countSize: '42',
            themeColor: '#219be0',
            totalCountNumber: this.feedbackCount.toString(),
          });
          this.stats.push({
            _id: 1,
            labelName: 'All',
            labelSize: '24',
            countNumber: this.totalFeedbackCount.toString(),
            countSize: '42',
            themeColor: '#219be0',
            totalCountNumber: this.totalFeedbackCount.toString(),
          });
```

File: packages/feedback-spa/src/app/feedback/home/feedback-home.component.ts [hunk 9]
```diff
@@ -34,140 +33,165 @@ export class FeedbackHomeComponent implements OnInit {
...
+          }
+        }
+      );
+    });
+    this.feedbackTags.push({
       _id: 1,
-      state: 'All'
-    } );
-    this.feedbackTags.push( {
+      state: "All",
```
Comment (riginoommen) [inline packages/feedback-spa/src/app/feedback/home/feedback-home.component.ts:113]:
```suggestion
      state: 'All',
```

File: packages/feedback-spa/src/app/feedback/home/feedback-home.component.ts [hunk 10]
```diff
@@ -34,140 +33,165 @@ export class FeedbackHomeComponent implements OnInit {
...
+    });
+    this.feedbackTags.push({
       _id: 2,
-      state: 'Open',
-    } );
-    this.feedbackTags.push( {
+      state: "Open",
```
Comment (riginoommen) [inline packages/feedback-spa/src/app/feedback/home/feedback-home.component.ts:117]:
```suggestion
      state: 'Open',
```

File: packages/feedback-spa/src/app/feedback/home/feedback-home.component.ts [hunk 11]
```diff
@@ -34,140 +33,165 @@ export class FeedbackHomeComponent implements OnInit {
...
+    });
+    this.feedbackTags.push({
       _id: 3,
-      state: 'Close'
-    } );
+      state: "Close",
```
Comment (riginoommen) [inline packages/feedback-spa/src/app/feedback/home/feedback-home.component.ts:121]:
```suggestion
      state: 'Close',
```

File: packages/feedback-spa/src/app/feedback/home/feedback-home.component.ts [hunk 12]
```diff
@@ -34,140 +33,165 @@ export class FeedbackHomeComponent implements OnInit {
...
+  setTypeFilter(type, index) {
     this.selectedIndex = index;
-    if ( type === 'Feedback' ) {
-      this.feedbackType = 'Feedback';
-    } else if ( type === 'Bugs' ) {
-      this.feedbackType = 'Bug';
+    if (type === "Feedback") {
+      this.feedbackType = "Feedback";
+    } else if (type === "Bugs") {
+      this.feedbackType = "Bug";
```
Comment (riginoommen) [inline packages/feedback-spa/src/app/feedback/home/feedback-home.component.ts:129]:
```suggestion
    if (type === 'Feedback') {
      this.feedbackType = 'Feedback';
    } else if (type === 'Bugs') {
      this.feedbackType = 'Bug';
```

File: packages/feedback-spa/src/app/feedback/home/feedback-home.component.ts [hunk 13]
```diff
@@ -34,140 +33,165 @@ export class FeedbackHomeComponent implements OnInit {
...
+  setActive(state, index) {
     this.selectedTab = index;
-    if ( state === 'Open' ) {
-      this.tagState = 'opened';
-    } else if ( state === 'Close' ) {
-      this.tagState = 'closed';
+    if (state === "Open") {
+      this.tagState = "opened";
+    } else if (state === "Close") {
+      this.tagState = "closed";
```
Comment (riginoommen) [inline packages/feedback-spa/src/app/feedback/home/feedback-home.component.ts:140]:
```suggestion
    if (state === 'Open') {
      this.tagState = 'opened';
    } else if (state === 'Close') {
      this.tagState = 'closed';
```

File: packages/feedback-spa/src/app/feedback/home/feedback-home.component.ts [hunk 14]
```diff
@@ -34,140 +33,165 @@ export class FeedbackHomeComponent implements OnInit {
...
-  setFeedbackCount () {
-    if ( this.feedbackType ) {
-      if ( this.moduleName ) {
+  setFeedbackCount() {
+    if (this.feedbackType) {
+      if (this.moduleName) {
         this.feedbackOpenedCount = this.allFeedback.filter(
-          fb => fb.state === 'opened' && fb.feedbackType === this.feedbackType && fb.module.name === this.moduleName ).length;
+          (fb) =>
+            fb.state === "opened" &&
```
Comment (riginoommen) [inline packages/feedback-spa/src/app/feedback/home/feedback-home.component.ts:150]:
```suggestion
            fb.state === 'opened' &&
```

**Comment by hybridx** [comment]
> @hybridx I kept trying to use `'` but I think `editorconfig` conf kept overwriting it. Need to check that in the future.

If you individually open up the SPA in code it allows you to edit I suppose

File: packages/feedback-spa/src/app/feedback/home/feedback-home.component.ts [hunk 15]
```diff
@@ -34,140 +33,165 @@ export class FeedbackHomeComponent implements OnInit {
...
+            fb.feedbackType === this.feedbackType &&
+            fb.module.name === this.moduleName
+        ).length;
         this.feedbackClosedCount = this.allFeedback.filter(
-          fb => fb.state !== 'opened' && fb.feedbackType === this.feedbackType && fb.module.name === this.moduleName ).length;
+          (fb) =>
+            fb.state !== "opened" &&
```
Comment (riginoommen) [inline packages/feedback-spa/src/app/feedback/home/feedback-home.component.ts:156]:
```suggestion
            fb.state !== 'opened' &&
```

File: packages/feedback-spa/src/app/feedback/home/feedback-home.component.ts [hunk 16]
```diff
@@ -34,140 +33,165 @@ export class FeedbackHomeComponent implements OnInit {
...
+        ).length;
       } else {
-        this.feedbackOpenedCount = this.allFeedback.filter( fb => fb.state === 'opened' && fb.feedbackType === this.feedbackType ).length;
-        this.feedbackClosedCount = this.allFeedback.filter( fb => fb.state !== 'opened' && fb.feedbackType === this.feedbackType ).length;
+        this.feedbackOpenedCount = this.allFeedback.filter(
+          (fb) => fb.state === "opened" && fb.feedbackType === this.feedbackType
+        ).length;
+        this.feedbackClosedCount = this.allFeedback.filter(
+          (fb) => fb.state !== "opened" && fb.feedbackType === this.feedbackType
+        ).length;
```
Comment (riginoommen) [inline packages/feedback-spa/src/app/feedback/home/feedback-home.component.ts:166]:
```suggestion
          (fb) => fb.state === 'opened' && fb.feedbackType === this.feedbackType
        ).length;
        this.feedbackClosedCount = this.allFeedback.filter(
          (fb) => fb.state !== 'opened' && fb.feedbackType === this.feedbackType
        ).length;
```

File: packages/feedback-spa/src/app/feedback/home/feedback-home.component.ts [hunk 17]
```diff
@@ -34,140 +33,165 @@ export class FeedbackHomeComponent implements OnInit {
...
-      if ( this.moduleName ) {
-        this.feedbackOpenedCount = this.allFeedback.filter( fb => fb.state === 'opened' && fb.module.name === this.moduleName ).length;
-        this.feedbackClosedCount = this.allFeedback.filter( fb => fb.state !== 'opened' && fb.module.name === this.moduleName ).length;
+      if (this.moduleName) {
+        this.feedbackOpenedCount = this.allFeedback.filter(
+          (fb) => fb.state === "opened" && fb.module.name === this.moduleName
+        ).length;
+        this.feedbackClosedCount = this.allFeedback.filter(
+          (fb) => fb.state !== "opened" && fb.module.name === this.moduleName
+        ).length;
```
Comment (riginoommen) [inline packages/feedback-spa/src/app/feedback/home/feedback-home.component.ts:175]:
```suggestion
      if (this.moduleName) {
        this.feedbackOpenedCount = this.allFeedback.filter(
          (fb) => fb.state === 'opened' && fb.module.name === this.moduleName
        ).length;
        this.feedbackClosedCount = this.allFeedback.filter(
          (fb) => fb.state !== 'opened' && fb.module.name === this.moduleName
        ).length;
```

File: packages/feedback-spa/src/app/feedback/home/feedback-home.component.ts [hunk 18]
```diff
@@ -34,140 +33,165 @@ export class FeedbackHomeComponent implements OnInit {
...
       } else {
-        this.feedbackOpenedCount = this.allFeedback.filter( fb => fb.state === 'opened' ).length;
-        this.feedbackClosedCount = this.allFeedback.filter( fb => fb.state !== 'opened' ).length;
+        this.feedbackOpenedCount = this.allFeedback.filter(
+          (fb) => fb.state === "opened"
+        ).length;
+        this.feedbackClosedCount = this.allFeedback.filter(
+          (fb) => fb.state !== "opened"
+        ).length;
```
Comment (riginoommen) [inline packages/feedback-spa/src/app/feedback/home/feedback-home.component.ts:182]:
```suggestion
        this.feedbackOpenedCount = this.allFeedback.filter(
          (fb) => fb.state === 'opened'
        ).length;
        this.feedbackClosedCount = this.allFeedback.filter(
          (fb) => fb.state !== 'opened'
        ).length;
```

File: packages/feedback-spa/src/app/feedback/home/feedback-home.component.ts [hunk 19]
```diff
@@ -34,140 +33,165 @@ export class FeedbackHomeComponent implements OnInit {
...
       }
     }
   }
-  filterFeedback ( feedbackModule ) {
-    if ( feedbackModule === 'Any' ) {
+  filterFeedback(feedbackModule) {
+    if (feedbackModule === "Any") {
```
Comment (riginoommen) [inline packages/feedback-spa/src/app/feedback/home/feedback-home.component.ts:187]:
```suggestion
    if (feedbackModule === 'Any') {
```

File: packages/feedback-spa/src/environments/environment.ts [hunk 20]
```diff
@@ -5,12 +5,8 @@
...
-  jiraAPI: '',
-  graphqlAPI: '',
-  client: '',
-  authorization: '',
-  jiraURL: '',
-  roverURL: '',
+  name: "local",
+  graphqlAPI: "",
+  jiraURL: "",
+  userProfileURL: "",
```
Comment (riginoommen) [inline packages/feedback-spa/src/environments/environment.ts:11]:
```suggestion
  name: 'local',
  graphqlAPI: '',
  jiraURL: '',
  userProfileURL: '',
```

File: packages/feedback-spa/src/environments/environment.beta.ts [hunk 21]
```diff
@@ -1,16 +1,12 @@
...
-  jiraAPI: '',
-  graphqlAPI: '',
-  client: '',
-  authorization: '',
-  jiraURL: '',
-  roverURL: '',
+  name: "beta",
+  graphqlAPI: "",
+  jiraURL: "",
+  userProfileURL: "",
```
Comment (riginoommen) [inline packages/feedback-spa/src/environments/environment.beta.ts:11]:
```suggestion
  name: 'beta',
  graphqlAPI: '',
  jiraURL: '',
  userProfileURL: '',
```

File: packages/feedback-spa/src/environments/environment.prod.ts [hunk 22]
```diff
@@ -1,11 +1,7 @@
...
-  api: '',
-  jiraAPI: '',
-  graphqlAPI: '',
-  client: '',
-  authorization: '',
-  jiraURL: '',
-  roverURL: '',
+  name: "",
+  graphqlAPI: "",
+  jiraURL: "",
```
Comment (riginoommen) [inline packages/feedback-spa/src/environments/environment.prod.ts:5]:
```suggestion
  name: '',
  graphqlAPI: '',
  jiraURL: '',
```

File: packages/feedback-spa/src/environments/environment.prod.ts [hunk 23]
```diff
@@ -1,11 +1,7 @@
...
+  userProfileURL: "",
```
Comment (riginoommen) [inline packages/feedback-spa/src/environments/environment.prod.ts:6]:
```suggestion
  userProfileURL: '',
```

File: packages/feedback-spa/src/environments/environment.qa.ts [hunk 24]
```diff
@@ -5,12 +5,8 @@
...
-  jiraAPI: '',
-  graphqlAPI: '',
-  client: '',
-  authorization: '',
-  jiraURL: '',
-  roverURL: '',
+  name: "qa",
+  graphqlAPI: "",
+  jiraURL: "",
+  userProfileURL: "",
```
Comment (riginoommen) [inline packages/feedback-spa/src/environments/environment.qa.ts:11]:
```suggestion
  name: 'qa',
  graphqlAPI: '',
  jiraURL: '',
  userProfileURL: '',
```

File: packages/feedback-spa/src/app/feedback/feedback-item/feedback-item.component.ts [hunk 25]
```diff
@@ -1,21 +1,19 @@
-import { Component, OnInit, Input, Output, EventEmitter} from '@angular/core';
+import { Component, OnInit, Input, Output, EventEmitter } from "@angular/core";
```
Comment (riginoommen) [inline packages/feedback-spa/src/app/feedback/feedback-item/feedback-item.component.ts:1]:
```suggestion
import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
```

**Comment by sayak-sarkar** [comment]
Created #484 with an enhanced deployment script and repository cleanup chores.

**Pull request closed without merging**